### PR TITLE
Add rpcinfo feature

### PR DESCRIPTION
**Pull request by LionZXY**
**Background**

This PR is one part of a larger effort to migrate all the old code to the new BLE API.
The goal of this PR is to migrate rpc info

**Changes**

- Add transport meta information API
- Add version feature API
- Add rpc info API
- Migrate manual DI factories map to qualifier multibindings Dagger DI
- Add getinfo feature api
- Add to feature factory methods suspend modifier

**Test plan**

Open sample app and press on "Get rpc info"

### CODE REVIEW

File: components/bridge/connection/device/fzero/impl/src/main/kotlin/com/flipperdevices/bridge/connection/device/fzero/impl/api/FZeroDeviceApiImpl.kt [hunk 1]
```diff
@@ -25,20 +26,31 @@ import kotlin.reflect.KClass
...
+    onReadyFeaturesApiFactories: MutableSet<FOnDeviceReadyFeatureApi.Factory>,
+    private val factories: MutableMap<FDeviceFeature, FDeviceFeatureApi.Factory>
 ) : FZeroDeviceApi, FUnsafeDeviceFeatureApi, LogTagProvider {
     override val TAG = "FZeroDeviceApi"
 
     private val features = EnumMap<FDeviceFeature, FDeviceFeatureApi>(FDeviceFeature::class.java)
     private val mutex = Mutex()
+
+    init {
+        if (BuildConfig.DEBUG) {
```
Comment (makeevrserg) [inline components/bridge/connection/device/fzero/impl/src/main/kotlin/com/flipperdevices/bridge/connection/device/fzero/impl/api/FZeroDeviceApiImpl.kt:38]: maybe use multiplatform BuildKonfig.IS_LOG_ENABLED?

File: components/bridge/connection/feature/getinfo/impl/src/main/kotlin/com/flipperdevices/bridge/connection/feature/getinfo/impl/mapper/DeviceInfoKeyMapper.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.flipperdevices.bridge.connection.feature.getinfo.impl.mapper
+
+import com.flipperdevices.bridge.connection.feature.getinfo.model.FGetInfoApiGroup
+import com.flipperdevices.bridge.connection.feature.getinfo.model.FGetInfoApiProperty
+
+class DeviceInfoKeyMapper {
+    private val cache = FGetInfoApiProperty.DeviceInfo.entries.associateBy {
+        it.key
+    }
+
```
Comment (makeevrserg) [inline components/bridge/connection/feature/getinfo/impl/src/main/kotlin/com/flipperdevices/bridge/connection/feature/getinfo/impl/mapper/DeviceInfoKeyMapper.kt:10]: NIT: Maybe something like this?
```kotlin
    private val cache = FGetInfoApiProperty.DeviceInfo
        .entries
// here by reference
        .associateBy(FGetInfoApiProperty.DeviceInfo::key)
 
    // use getOrElse
    fun map(path: String): FGetInfoApiProperty {
        return cache.getOrElse(path) {
            FGetInfoApiProperty.Unknown(FGetInfoApiGroup.DEVICE_INFO, path)
        }
    }
```

File: components/bridge/connection/feature/protocolversion/impl/src/main/kotlin/com/flipperdevices/bridge/connection/feature/protocolversion/impl/api/FVersionFeatureApiImpl.kt [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+class FVersionFeatureApiImpl @AssistedInject constructor(
+    settingsStoreProvider: Provider<DataStore<Settings>>,
+    @Assisted private val scope: CoroutineScope,
+    @Assisted metaInfoApi: FTransportMetaInfoApi
+) : FVersionFeatureApi, LogTagProvider {
+    override val TAG = "FlipperVersionApi"
+
+    private val ignoreUnsupported by lazy {
+        runBlocking {
+            settingsStoreProvider.get().data.first().ignoreUnsupportedVersion
```
Comment (makeevrserg) [inline components/bridge/connection/feature/protocolversion/impl/src/main/kotlin/com/flipperdevices/bridge/connection/feature/protocolversion/impl/api/FVersionFeatureApiImpl.kt:44]: We have scope here, is it required to use runblocking right here? May be possible to launch it in init?

**Review by makeevrserg — APPROVED** [review]

File: components/bridge/connection/feature/getinfo/impl/src/main/kotlin/com/flipperdevices/bridge/connection/feature/getinfo/impl/mapper/DeviceInfoKeyMapper.kt (hunk 2 above)
Comment (LionZXY) [reply components/bridge/connection/feature/getinfo/impl/src/main/kotlin/com/flipperdevices/bridge/connection/feature/getinfo/impl/mapper/DeviceInfoKeyMapper.kt:10]: I don't want to store unknown property keys and thus take up extra RAM

**Review by makeevrserg — APPROVED** [review]

**Pull request merged**